### DISCUSSION TITLE
How is GPTCache different from Pinecone

### DISCUSSION THREAD

**Discussion by yhyu13** [zilliztech/GPTCache#321] [Q&A] [answered]
Pinecone is a database that can be used to add long-term memory to AutoGPT, an experimental GPT-4 project[1](https://www.mlq.ai/long-term-memory-autogpt-pinecone/)[2](https://www.reddit.com/r/AutoGPT/comments/12kcit6/pinecone_never_seems_to_work_tips/). It is one of the memory backends that can be used with AutoGPT, along with local cache and Redis

Correct me if I am wrong, it seems to me GPTCache is like a specialized implementation of Pinecone in terms of LLM. 

So GPTCache is indeed a vector database with more than queries on top of it?

**Comment by SimFG** [comment, marked as answer]
In the AutoGPT project, the memory module is used to construct the context when generating the next command, which is more suitable  to be called the cache. The GPTCahce is not a vector store, and it's fitter to the AutoGPT, because it can provide more function, like costuming embedding  model, more data management, more accurate context info and so on. Of course, you can try to know about our vector store, Milvus. If  you want to quickly experience it, you can visit the [zilliz cloud](https://cloud.zilliz.com/).

**Reply by yhyu13** [reply]
lol, just found a blog that describe the origin of GPTCache project, this is quite profound, and a good piece of material to describe this project, too
https://www.modb.pro/db/623770
you guys should put this in the README
